src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -1326,7 +1326,8 @@ private RangeStreamer getRebuildStreamer(String sourceDc, String keyspace)
                                                               DatabaseDescriptor.getEndpointSnitch().getRack(address)));
             String localDc = DatabaseDescriptor.getEndpointSnitch().getDatacenter(FBUtilities.getBroadcastAddress());
             String localRack = DatabaseDescriptor.getEndpointSnitch().getRack(FBUtilities.getBroadcastAddress());
-            streamer.addSourceFilter(SingleRackFilter.create(topology, sourceDc, localDc, localRack));
+            AbstractReplicationStrategy replicationStrategy = Keyspace.open(keyspace).getReplicationStrategy();
```

Comment:
what happens if a rebuild is triggered without specifying any keyspace? will this break here